tools/distpartitioning/dataset_utils.py

```diff
@@ -154,25 +129,63 @@ def get_dataset(input_dir, graph_name, rank, world_size, schema_map):
-                                    num_chunks=world_size)
-    for ntype_name in schema_map[constants.STR_NODE_TYPE]: 
-        if ntype_name in node_feature_tids: 
-            for item in node_feature_tids[ntype_name]:
-                item[1] = node_tids[ntype_name][rank][0]
-                item[2] = node_tids[ntype_name][rank][1]
+                                    num_chunks=num_parts)
+
+    #iterate over the "node_data" dictionary in the schema_map
+    #read the node features if exists
+    #also keep track of the type_nids for which the node_features are read.
+    dataset_features = schema_map[constants.STR_NODE_DATA]
+    if((dataset_features is not None) and (len(dataset_features) > 0)):
+        for ntype_name, ntype_feature_data in dataset_features.items():
+            for feat_name, feat_data in ntype_feature_data.items():
+                assert feat_data[constants.STR_FORMAT][constants.STR_NAME] == constants.STR_NUMPY
+
+                # It is guaranteed that num_chunks is always greater 
+                # than num_partitions. 
+                num_chunks = len(feat_data[constants.STR_DATA])
+                read_list = np.array_split(np.arange(num_chunks), num_parts)
+                for local_part_id in range(num_parts):
+                    if map_partid_rank(local_part_id, world_size) == rank:
+                        nfeat = []
+                        nfeat_tids = []
+                        for idx in read_list[local_part_id]:
+                            nfeat_file = feat_data[constants.STR_DATA][idx]
+                            if not os.path.isabs(nfeat_file):
+                                nfeat_file = os.path.join(input_dir, nfeat_file)
+                            logging.info(f'Loading node feature[{feat_name}] of ntype[{ntype_name}] from {nfeat_file}')
+                            nfeat.append(np.load(nfeat_file))
+                        nfeat = np.concatenate(nfeat) if len(nfeat) != 0 else np.array([])
+                        node_features[ntype_name+"/"+feat_name+"/"+str(local_part_id//world_size)] = torch.from_numpy(nfeat)
+                        nfeat_tids.append(node_tids[ntype_name][local_part_id])
+                        node_feature_tids[ntype_name+"/"+feat_name+"/"+str(local_part_id//world_size)] = nfeat_tids
 
     #done building node_features locally. 
     if len(node_features) <= 0:
         logging.info(f'[Rank: {rank}] This dataset does not have any node features')
     else:
-        for k, v in node_features.items():
-            logging.info(f'[Rank: {rank}] node feature name: {k}, feature data shape: {v.size()}')
+        assert len(node_features) == len(node_feature_tids)
+
+        for feat_name, feat_info  in node_features.items():
+            logging.info(f'[Rank: {rank}] node feature name: {feat_name}, feature data shape: {feat_info.size()}')
+
+            tokens = feat_name.split("/")
+            assert len(tokens) == 3
+            strip_feat_name = "/".join(tokens[:-1])
```

Comment:
add a comment as it looks a bit hacky. show what `feat_name` looks like.

Reply:
this was unused code and hence deleted. however I added a comment to describe the format for the keys in this dictionary.